### PR TITLE
[FLINK-23739][table]BlackHoleSink & PrintSink implement SupportsParti…

### PR DESCRIPTION
**Pull request by YesOrNo828**
…tioning interface

## What is the purpose of the change

- Print & BlaskHole connector supoort partitioned table.

## Brief change log

- Inner class PrintSink implement SupportsPartitioning interface
- Inner class BlackHoleSink implement SupportsPartitioning interface

## Verifying this change

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? no

### CODE REVIEW

File: flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/connector/print/table/PrintTableSinkFactory.java
Comment (JingsongLi) [inline flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/connector/print/table/PrintTableSinkFactory.java:133]: I think it is better to print to StaticPartition, we can append it to printIdentifier.

**Review by JingsongLi — CHANGES_REQUESTED** [review]
Thanks @YesOrNo828 , we can add tests in `TableSinkTest`.

File: flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/connector/print/table/PrintTableSinkFactory.java
Comment (YesOrNo828) [reply flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/connector/print/table/PrintTableSinkFactory.java]: Done for print StaticPartition, and added some tests in `PrintTableSinkFactoryTest.java` & `BlackHoleSinkFactoryTest.java`.

**Comment by YesOrNo828** [comment]
@JingsongLi  Thank you for your reviewing. Is there anything I should do with this pr before merged?
BTW,  should I take care of the status of failure with CI report ? It seemed **test_ci kafka_gelly** has failed.

File: flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/connector/print/table/PrintTableSinkFactory.java
Comment (tsreaper) [inline flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/connector/print/table/PrintTableSinkFactory.java:132]: These changes are not covered. Consider adding a test in `PrintConnectorITCase` so that the result shows this `printIdentifier`.

**Comment by tsreaper** [comment]
Also please do not squash changes when updating this PR. Just push them directly so that the reviewers can see the changes. The committer who merges this commit will squash all changes for you when merging.

**Comment by YesOrNo828** [comment]
> Also please do not squash changes when updating this PR. Just push them directly so that the reviewers can see the changes. The committer who merges this commit will squash all changes for you when merging.

Thanks for your reminding.

File: flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/connector/print/table/PrintTableSinkFactory.java
Comment (YesOrNo828) [reply flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/connector/print/table/PrintTableSinkFactory.java:132]: Let me add more tests.

**Review by JingsongLi — COMMENTED** [review]
Thanks for your update, @YesOrNo828 Looks good to me.
At the moment of releasing 1.14, according to the rules of the community, I will wait until the 1.14 branch is pulled out, and then merge this PR.

**Review by JingsongLi — APPROVED** [review]
Looks good to me! Thanks @YesOrNo828

**Pull request merged**
